app/scripts/modules/core/src/pipeline/config/triggers/pubsub/pubsub.trigger.ts

```diff
@@ -1,19 +1,29 @@
 import { IController, module } from 'angular';
 
 import { PIPELINE_CONFIG_PROVIDER, PipelineConfigProvider } from 'core/pipeline/config/pipelineConfigProvider';
-import { IPubsubTrigger } from '@spinnaker/core';
+import {
+  IPubsubTrigger,
+  PUBSUB_SUBSCRIPTION_SERVICE,
+  PubsubSubscriptionService,
+} from '@spinnaker/core';
 
 class PubsubTriggerController implements IController {
   public pubsubSystems = ['kafka', 'google'];
+  public pubsubSubscriptions: string[];
 
-  constructor(public trigger: IPubsubTrigger) {
+  constructor(public trigger: IPubsubTrigger,
+              public pubsubSubscriptionService: PubsubSubscriptionService) {
     'ngInject';
+    pubsubSubscriptionService.getPubsubSubscriptions().then(subscriptions => {
```

Comment:
Ideally the full set of states of this call would be surfaced to the user. We generally don't end up surfacing failure, but you should show a spinner that indicates that the subscriptions are still loading. See https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/pipeline/config/triggers/jenkins/jenkinsTrigger.module.js#L89-L98

For a failed call, you can add `.catch(() => this.pubsubSubscriptions = []);`

Reply:
Done.